### PR TITLE
fix(QSelect): accept object type as modelValue (fix: #11485)

### PR DESCRIPTION
**Pull request by Evertvdw**
@yusufkandemir 
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [x] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch (or `v[X]` branch)
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [x] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

### CODE REVIEW

File: ui/src/components/select/QSelect.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "props": {
     "model-value": {
       "extends": "model-value",
-      "type": [ "Number", "String", "Array", "null", "undefined" ],
+      "type": [ "Number", "String", "Array", "Object", "null", "undefined" ],
```
Comment (yusufkandemir) [inline ui/src/components/select/QSelect.json]: I just figured we are using `Any` in all the other places such as return type of `option-value` prop, `opt` references in scopes of slots, etc. even `@update:model-value` has `Any` implicitly. Also, logically there is nothing holding the users back from using bigint, symbol, etc. too. So, I think we can just simply use the type `Any` here.
```suggestion
      "type": "Any",
```

**Review by yusufkandemir — CHANGES_REQUESTED** [review]
Thanks 👏 I suggested a change that is a fundamentally different solution. So, please consider it instead, if you don't have anything else to add.

**Comment by Evertvdw** [comment]
Fine with me! I guess any will do in this case 👍

**Review by yusufkandemir — APPROVED** [review]
Looks better now 👏

**Comment by rstoenescu** [comment]
Thanks for contributing!

**Pull request merged**